app/components/elements/forms/field_component.rb

```diff
@@ -37,6 +37,12 @@ def help_text_id
       end
 
       def field_aria
+        {}.tap do |arias|
+          arias[:required] = true if required
+        end.merge(error_aria)
```

Comment:
does this just set `aria-required` if the field is required? that would have no effect, if so – `aria-required` only exists for cases where you cannot set the ordinary required attribute. if `required` is set, the browser or SR will already know that without the need for aria.

Reply:
@thatbudakguy right, the bigger issue here is that setting `required` means the form will not submit, but we do validation on the server side not the client side to avoid javascript for this.

Reply:
it's totally OK (imo) to just set `aria-required` everywhere we would otherwise want to set `required`. the point is that you never need both. `required` has behavior we don't want (it blocks submission of the form), so all we need to do is set `aria-required` to convey the same thing without blocking submission.

Reply:
@thatbudakguy I feel like I'm missing something, isn't that what this is doing then? Setting `aria-required: true` when we pass `required: true` to the component? This PR removes `required` from the text field rendered by the component in favor of `aria-required`...

Reply:
Oh, then I misunderstood! I thought it additionally set aria-required everywhere there was already required, instead of replacing.